pkg/auth/metrics.go

```diff
@@ -153,3 +154,11 @@ func StatusLabel(status int) prometheus.Labels {
 	labels[StatusCode] = strconv.Itoa(status)
 	return labels
 }
+
+func UpstreamAuthLabel(labels prometheus.Labels, hasUpstreamAuth string) prometheus.Labels {
```

Comment:
the idea of having "prometheus.labels" as an input is very cool @Cypherspark .
but I think we should do followings in order to have consistent coding style:

- change the name to `AddUpstreamAuthMetricLabel`
- change `ReasonLabel` function to match this implementation

Reply:
Done.
yeah i was thinking something like a builder pattern for them but as prom.labels wan't defined locally it was not possible, so i decided to get them as input.